app/api/odm/model.ts

```diff
@@ -28,33 +46,48 @@ class UpdateLogHelper {
     );
   }
 
-  async upsertLogMany(affectedIds: any[], deleted = false) {
-    await updatelogsModel._updateMany(
-      { mongoId: { $in: affectedIds }, namespace: this.collectionName },
-      { $set: { timestamp: Date.now(), deleted } },
-      { lean: true }
-    );
+  async upsertLogMany(
+    query: UwaziFilterQuery<T>,
+    deleted = false,
+    batchSize = UpdateLogHelper.batchSizeUpsertMany
+  ) {
+    await new PromisePool()
+      .for(await getBatchSteps(models[this.collectionName], query, batchSize))
+      .withConcurrency(5)
+      .process(async (stepIndex: T) => {
+        const batch = await models[this.collectionName].get(
+          { ...query, $and: [{ _id: { $gte: stepIndex } }] },
+          { _id: 1 },
+          { limit: batchSize }
+        );
+
+        await updatelogsModel._updateMany(
+          { mongoId: { $in: batch }, namespace: this.collectionName },
+          { $set: { timestamp: Date.now(), deleted } },
```

Comment:
Is this true that an entire batch is either deleted true or deleted false?  Seems like a very weird approach.  I understand that this is also on the original code, that is why I have doubt.

Reply:
yes this is true, we have no operations that use this hooks with a mixed update/create/delete, actually i do not know if this is possible using mongo either but definitely not our case.